components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -133,7 +152,12 @@ class GeckoEngineSession(
      * See [EngineSession.reload]
      */
     override fun reload(flags: LoadUrlFlags) {
-        geckoSession.reload(flags.value)
+        val loadRequest = pendingLoadRequest
+        if (loadRequest != null) {
+            loadUrl(loadRequest.url, loadRequest.parent, loadRequest.flags, loadRequest.additionalHeaders)
```

Comment:
I wonder if we should do this only for the first load? If I am on site A and load site B, stop the load, remain on site A and press reload then this would load site B now? Maybe this is only a problem for the `about:blank` load that no one wanted?

Reply:
@pocmo ah yes, definitely! I pushed an update that handles this now and adds comments and tests.